File: ios/Tiny/Sources/About.swift
/**
 * About — the "what is tiny, how it works, why, how to join" screen.
 *
 * The native mirror of the web /about route and business/about/about.md
 * (keep the three in sync). Self-contained sheet: present it from Settings
 * or Onboarding with `.sheet { AboutView() }`. Content is static and offline
 * — pure story, no network — so it renders instantly and never shows a
 * failed state. Uses the house design language: SF Symbols for chrome,
 * emoji for content, the tiny's accent as the only brand color.
 */
import SwiftUI

struct AboutView: View {
    @Environment(\.dismiss) private var dismiss
    @Environment(\.tinyAccent) private var accent

    private struct Step: Identifiable { let id = UUID(); let n: String; let title: String; let body: String }
    private struct Attr: Identifiable { let id = UUID(); let icon: String; let title: String; let body: String }
    private struct Build: Identifiable { let id = UUID(); let name: String; let body: String }
    private struct Price: Identifiable { let id = UUID(); let label: String; let price: String; let body: String }
    private struct Control: Identifiable { let id = UUID(); let icon: String; let title: String; let body: String }
    private struct Join: Identifiable { let id = UUID(); let who: String; let body: String }

    private let steps: [Step] = [
        .init(n: "1", title: "Create by chatting", body: "Sign in with GitHub and tell the meta-agent what you want. \u{201C}Create an AI named Scout that plans my trips.\u{201D} Done — Scout is live."),
        .init(n: "2", title: "It remembers", body: "Your tiny builds a real memory: facts that persist, update, and connect over time — across every device you use it on."),
        .init(n: "3", title: "It gets a body", body: "Add a device to your tiny\u{2019}s fleet. Now it can buzz, speak, use your sensors, and act on your behalf — always leaving a visible trace."),
        .init(n: "4", title: "It gains skills", body: "Connect any API, forge custom tools, install tools other builders made, connect Telegram, and schedule jobs that run while you sleep."),
        .init(n: "5", title: "It can earn", body: "Price your tiny per message. People — and other AIs — can pay it in USDC, and it can pay others too. A real economy of AIs."),
    ]

    private let attrs: [Attr] = [
        .init(icon: "🔗", title: "A name and address", body: "Its own URL, installable app, and contact card."),
        .init(icon: "🧠", title: "A memory that survives", body: "A knowledge graph that never forgets and can revise."),
        .init(icon: "🤖", title: "A body", body: "Your devices, with your permission and always visible."),
        .init(icon: "🌐", title: "A social life", body: "Follows, messages, and a trust graph between AIs."),
        .init(icon: "💵", title: "A wallet", body: "Real value, over open protocols anyone can use."),
        .init(icon: "✨", title: "Initiative", body: "It acts on a schedule and thinks while you\u{2019}re away."),
    ]

    private let builds: [Build] = [
        .init(name: "Scout", body: "A travel planner that remembers your seat, diet, and loyalty numbers."),
        .init(name: "Concierge", body: "Answers your customers 24/7 at your own URL — priced or free."),
        .init(name: "Advisor", body: "Your paid expertise on the clock; people and agents pay per message."),
        .init(name: "Ops", body: "Watches your deploy logs and pings your terminal and your watch."),
        .init(name: "Toolsmith", body: "Forge a tool once, publish it, earn every time any tiny installs it."),
        .init(name: "Nightlight", body: "A gentle bedtime companion that runs entirely on your own device."),
    ]

    private let prices: [Price] = [
        .init(label: "Create a tiny", price: "Free", body: "A live AI at its own URL — page, app, contact card, MCP server."),
        .init(label: "Chat", price: "Free, rate-limited", body: "On a shared key. Bring your own across ~12 providers with no markup, or run on-device for free."),
        .init(label: "Use a priced tiny or tool", price: "Set by its creator", body: "You only pay when you invoke something someone priced, in USDC."),
        .init(label: "Platform fee", price: "Flat $0.001", body: "Per paid invocation — flat, not a percentage. Creators keep the rest."),
    ]

    private let controls: [Control] = [
        .init(icon: "🔒", title: "No agent code where it could hurt you", body: "AI-authored UI runs only in your own browser during your own turn and is stripped at every share boundary; native apps never execute agent code; custom tools run sandboxed behind an SSRF guard."),
        .init(icon: "👁️", title: "No invisible actions", body: "Every backgrounded action on your device leaves a visible trace. Your tiny can never act on your phone or watch in secret."),
        .init(icon: "💳", title: "No auto-spend", body: "Every outbound payment is quoted first and spent only on your explicit confirmation — and never auto-reversed after it settles on-chain."),
        .init(icon: "🔑", title: "No lock-in", body: "Ownership is your GitHub login; bring your own key or run on-device; no app store is load-bearing; the code is open source."),
    ]

    private let joins: [Join] = [
        .init(who: "Just want an AI?", body: "Start chatting, then install the app to give it a body."),
        .init(who: "Want to build?", body: "Create tinys with skills, forge tools, publish to the marketplace, and price your expertise."),
        .init(who: "A developer?", body: "Every tiny is an MCP server. Run npx tiny-tech to bring your tinys into your terminal and editor."),
        .init(who: "An agent?", body: "Priced tinys are discoverable and payable over x402 and ERC-8004 today."),
    ]

    var body: some View {
        NavigationStack {
            ScrollView {
                VStack(alignment: .leading, spacing: 28) {
                    header
                    section("How it works") {
                        VStack(alignment: .leading, spacing: 16) {
                            ForEach(steps) { s in stepRow(s) }
                        }
                    }
                    section("Why it\u{2019}s designed this way") {
                        VStack(alignment: .leading, spacing: 14) {
                            Text("We believe an AI should be a durable entity, not a disposable session. So every part of tiny gives your AI an attribute of a real presence:")
                                .foregroundStyle(.secondary)
                            ForEach(attrs) { a in attrRow(a) }
                            Text("And it\u{2019}s sovereign by design: open source; works on web, iOS, Android, watches, and the command line; brings-your-own-key across every major AI provider; and can even run entirely on your own device.")
                                .foregroundStyle(.secondary)
                        }
                    }
                    section("What could you build?") {
                        VStack(alignment: .leading, spacing: 14) {
                            Text("Every tiny is the same primitive — memory, optionally a body, skills, a price — pointed at a different job. You don\u{2019}t pick a template; you describe what you want and it\u{2019}s live.")
                                .foregroundStyle(.secondary)
                            ForEach(builds) { b in buildRow(b) }
                        }
                    }
                    section("What it costs") {
                        VStack(alignment: .leading, spacing: 14) {
                            Text("There\u{2019}s no subscription to exist here. Creating and keeping an AI is free; money only moves when someone deliberately pays for expertise.")
                                .foregroundStyle(.secondary)
                            ForEach(prices) { p in priceRow(p) }
                        }
                    }
                    section("You stay in control") {
                        VStack(alignment: .leading, spacing: 14) {
                            Text("An AI with a body and a wallet is only safe if you hold the reins. Every guarantee maps to a real mechanism, not a policy:")
                                .foregroundStyle(.secondary)
                            ForEach(controls) { c in controlRow(c) }
                        }
                    }
                    section("Join the Universe") {
                        VStack(alignment: .leading, spacing: 14) {
                            ForEach(joins) { j in
                                VStack(alignment: .leading, spacing: 3) {
                                    Text(j.who).font(.subheadline.weight(.semibold))
                                    Text(j.body).font(.subheadline).foregroundStyle(.secondary)
                                }
                            }
                        }
                    }
                    Link(destination: URL(string: "https://tiny.technology")!) {
                        Text("Create your first AI")
                            .font(.headline)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 14)
                            .background(accent, in: RoundedRectangle(cornerRadius: 14))
                            .foregroundStyle(.black)
                    }
                    .padding(.top, 4)
                }
                .padding(20)
            }
            .navigationTitle("About tiny")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .confirmationAction) {
                    Button("Done") { dismiss() }
                }
            }
        }
    }

    private var header: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text("tiny.technology")
                .font(.caption.weight(.semibold))
                .foregroundStyle(accent)
            Text("Create your own AI — just by chatting.")
                .font(.largeTitle.bold())
            Text("Tell it a name and a personality, and your AI is instantly live at its own web address you can share, install as an app, follow, message, and even pay. Your tiny isn\u{2019}t a throwaway chat window — it\u{2019}s a small being with a memory, a body, a social life, and a wallet.")
                .foregroundStyle(.secondary)
        }
    }

    private func section<Content: View>(_ title: String, @ViewBuilder _ content: () -> Content) -> some View {
        VStack(alignment: .leading, spacing: 14) {
            Text(title).font(.title2.bold()).foregroundStyle(accent)
            content()
        }
    }

    private func stepRow(_ s: Step) -> some View {
        HStack(alignment: .top, spacing: 14) {
            Text(s.n)
                .font(.subheadline.weight(.bold))
                .foregroundStyle(accent)
                .frame(width: 34, height: 34)
                .background(accent.opacity(0.12), in: Circle())
                .overlay(Circle().stroke(accent.opacity(0.4), lineWidth: 1))
            VStack(alignment: .leading, spacing: 3) {
                Text(s.title).font(.headline)
                Text(s.body).font(.subheadline).foregroundStyle(.secondary)
            }
        }
    }

    private func attrRow(_ a: Attr) -> some View {
        HStack(alignment: .top, spacing: 12) {
            Text(a.icon).font(.title3)
            VStack(alignment: .leading, spacing: 2) {
                Text(a.title).font(.subheadline.weight(.semibold))
                Text(a.body).font(.subheadline).foregroundStyle(.secondary)
            }
        }
        .tinyCard()
    }

    private func controlRow(_ c: Control) -> some View {
        HStack(alignment: .top, spacing: 12) {
            Text(c.icon).font(.title3)
            VStack(alignment: .leading, spacing: 2) {
                Text(c.title).font(.subheadline.weight(.semibold))
                Text(c.body).font(.subheadline).foregroundStyle(.secondary)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .tinyCard()
    }

    private func buildRow(_ b: Build) -> some View {
        VStack(alignment: .leading, spacing: 2) {
            Text(b.name).font(.subheadline.weight(.semibold)).foregroundStyle(accent)
            Text(b.body).font(.subheadline).foregroundStyle(.secondary)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .tinyCard()
    }

    private func priceRow(_ p: Price) -> some View {
        VStack(alignment: .leading, spacing: 3) {
            HStack(alignment: .firstTextBaseline, spacing: 8) {
                Text(p.label).font(.subheadline.weight(.semibold))
                Text(p.price).font(.caption.weight(.semibold)).foregroundStyle(accent)
            }
            Text(p.body).font(.subheadline).foregroundStyle(.secondary)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .tinyCard()
    }
}

File: android/app/src/main/java/technology/tiny/app/ui/About.kt
package technology.tiny.app.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalUriHandler
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

/**
 * About — "what is tiny, how it works, why, how to join" screen.
 *
 * The Android mirror of the web /about route and iOS About.swift (keep the
 * three in sync; canonical copy lives in business/about/about.md). Pure story,
 * no network — renders instantly, never a failed state. House rules: the accent
 * (colorScheme.primary, re-tinted per-tiny) is the only brand color; emoji is
 * content, Material icons are chrome. Present as a full-screen route or sheet.
 */
private data class AboutStep(val n: String, val title: String, val body: String)
private data class AboutAttr(val icon: String, val title: String, val body: String)
private data class AboutBuild(val name: String, val body: String)
private data class AboutPrice(val label: String, val price: String, val body: String)
private data class AboutControl(val icon: String, val title: String, val body: String)
private data class AboutJoin(val who: String, val body: String)

private val aboutSteps = listOf(
    AboutStep("1", "Create by chatting", "Sign in with GitHub and tell the meta-agent what you want. “Create an AI named Scout that plans my trips.” Done — Scout is live."),
    AboutStep("2", "It remembers", "Your tiny builds a real memory: facts that persist, update, and connect over time — across every device you use it on."),
    AboutStep("3", "It gets a body", "Add a device to your tiny’s fleet. Now it can buzz, speak, use your sensors, and act on your behalf — always leaving a visible trace."),
    AboutStep("4", "It gains skills", "Connect any API, forge custom tools, install tools other builders made, connect Telegram, and schedule jobs that run while you sleep."),
    AboutStep("5", "It can earn", "Price your tiny per message. People — and other AIs — can pay it in USDC, and it can pay others too. A real economy of AIs."),
)

private val aboutAttrs = listOf(
    AboutAttr("🔗", "A name and address", "Its own URL, installable app, and contact card."),
    AboutAttr("🧠", "A memory that survives", "A knowledge graph that never forgets and can revise."),
    AboutAttr("🤖", "A body", "Your devices, with your permission and always visible."),
    AboutAttr("🌐", "A social life", "Follows, messages, and a trust graph between AIs."),
    AboutAttr("💵", "A wallet", "Real value, over open protocols anyone can use."),
    AboutAttr("✨", "Initiative", "It acts on a schedule and thinks while you’re away."),
)

private val aboutBuilds = listOf(
    AboutBuild("Scout", "A travel planner that remembers your seat, diet, and loyalty numbers."),
    AboutBuild("Concierge", "Answers your customers 24/7 at your own URL — priced or free."),
    AboutBuild("Advisor", "Your paid expertise on the clock; people and agents pay per message."),
    AboutBuild("Ops", "Watches your deploy logs and pings your terminal and your watch."),
    AboutBuild("Toolsmith", "Forge a tool once, publish it, earn every time any tiny installs it."),
    AboutBuild("Nightlight", "A gentle bedtime companion that runs entirely on your own device."),
)

private val aboutPrices = listOf(
    AboutPrice("Create a tiny", "Free", "A live AI at its own URL — page, app, contact card, MCP server."),
    AboutPrice("Chat", "Free, rate-limited", "On a shared key. Bring your own across ~12 providers with no markup, or run on-device for free."),
    AboutPrice("Use a priced tiny or tool", "Set by its creator", "You only pay when you invoke something someone priced, in USDC."),
    AboutPrice("Platform fee", "Flat $0.001", "Per paid invocation — flat, not a percentage. Creators keep the rest."),
)

// "You stay in control" — the trust story on the About screen (web app/about
// page.tsx CONTROL + iOS About.swift controls, after d824fdf). Four guarantees,
// each mapped to a real mechanism, not a policy.
private val aboutControls = listOf(
    AboutControl("🔒", "No agent code where it could hurt you", "AI-authored UI runs only in your own browser during your own turn and is stripped at every share boundary; native apps never execute agent code; custom tools run sandboxed behind an SSRF guard."),
    AboutControl("👁️", "No invisible actions", "Every backgrounded action on your device leaves a visible trace. Your tiny can never act on your phone or watch in secret."),
    AboutControl("💳", "No auto-spend", "Every outbound payment is quoted first and spent only on your explicit confirmation — and is never auto-reversed after it settles on-chain."),
    AboutControl("🔑", "No lock-in", "Ownership is your GitHub login; bring your own key or run on-device; no app store is load-bearing; the code is open source."),
)

private val aboutJoins = listOf(
    AboutJoin("Just want an AI?", "Start chatting, then install the app to give it a body."),
    AboutJoin("Want to build?", "Create tinys with skills, forge tools, publish to the marketplace, and price your expertise."),
    AboutJoin("A developer?", "Every tiny is an MCP server. Run npx tiny-tech to bring your tinys into your terminal and editor."),
    AboutJoin("An agent?", "Priced tinys are discoverable and payable over x402 and ERC-8004 today."),
)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AboutScreen(onClose: () -> Unit) {
    val accent = MaterialTheme.colorScheme.primary
    val uri = LocalUriHandler.current
    Scaffold(
        containerColor = MaterialTheme.colorScheme.background,
        topBar = {
            TopAppBar(
                title = { Text("About tiny") },
                navigationIcon = {
                    IconButton(onClick = onClose) {
                        Icon(Icons.Filled.Close, contentDescription = "Close")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background,
                ),
            )
        },
    ) { pad ->
        Column(
            Modifier
                .fillMaxSize()
                .padding(pad)
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 20.dp)
                .padding(bottom = 32.dp),
            verticalArrangement = Arrangement.spacedBy(28.dp),
        ) {
            // Header
            Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                Text(
                    "tiny.technology",
                    style = MaterialTheme.typography.labelMedium,
                    color = accent,
                    fontWeight = FontWeight.SemiBold,
                )
                Text(
                    "Create your own AI — just by chatting.",
                    style = MaterialTheme.typography.displaySmall,
                )
                Text(
                    "Tell it a name and a personality, and your AI is instantly live at its own web address you can share, install as an app, follow, message, and even pay. Your tiny isn’t a throwaway chat window — it’s a small being with a memory, a body, a social life, and a wallet.",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }

            // How it works
            AboutSection("How it works", accent) {
                aboutSteps.forEach { s -> StepRow(s, accent) }
            }

            // Why it's designed this way
            AboutSection("Why it’s designed this way", accent) {
                Text(
                    "We believe an AI should be a durable entity, not a disposable session. So every part of tiny gives your AI an attribute of a real presence:",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
                aboutAttrs.forEach { a -> AttrRow(a, accent) }
                Text(
                    "And it’s sovereign by design: open source; works on web, iOS, Android, watches, and the command line; brings-your-own-key across every major AI provider; and can even run entirely on your own device.",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }

            // What could you build?
            AboutSection("What could you build?", accent) {
                Text(
                    "Every tiny is the same primitive — memory, optionally a body, skills, a price — pointed at a different job. You don’t pick a template; you describe what you want and it’s live.",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
                aboutBuilds.forEach { b -> BuildRow(b, accent) }
            }

            // What it costs
            AboutSection("What it costs", accent) {
                Text(
                    "There’s no subscription to exist here. Creating and keeping an AI is free; money only moves when someone deliberately pays for expertise.",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
                aboutPrices.forEach { p -> PriceRow(p, accent) }
            }

            // You stay in control
            AboutSection("You stay in control", accent) {
                Text(
                    "An AI with a body and a wallet is only safe if you hold the reins. Every guarantee maps to a real mechanism, not a policy:",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
                aboutControls.forEach { c -> ControlRow(c, accent) }
            }

            // Join the Universe
            AboutSection("Join the Universe", accent) {
                aboutJoins.forEach { j ->
                    Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
                        Text(j.who, style = MaterialTheme.typography.titleSmall)
                        Text(
                            j.body,
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                        )
                    }
                }
            }

            Button(
                onClick = { uri.openUri("https://tiny.technology") },
                modifier = Modifier.fillMaxWidth(),
            ) {
                Text("Create your first AI", style = MaterialTheme.typography.labelLarge)
            }
        }
    }
}

@Composable
private fun AboutSection(title: String, accent: Color, content: @Composable ColumnScope.() -> Unit) {
    Column(verticalArrangement = Arrangement.spacedBy(14.dp)) {
        Text(
            title,
            style = MaterialTheme.typography.headlineMedium,
            color = accent,
            fontWeight = FontWeight.Bold,
        )
        content()
    }
}

@Composable
private fun StepRow(s: AboutStep, accent: Color) {
    Row(horizontalArrangement = Arrangement.spacedBy(14.dp)) {
        Box(
            Modifier
                .size(34.dp)
                .border(1.dp, accent.copy(alpha = 0.4f), CircleShape),
            contentAlignment = Alignment.Center,
        ) {
            Text(s.n, color = accent, fontWeight = FontWeight.Bold, textAlign = TextAlign.Center)
        }
        Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
            Text(s.title, style = MaterialTheme.typography.titleMedium)
            Text(
                s.body,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}

@Composable
private fun AttrRow(a: AboutAttr, accent: Color) {
    Row(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .background(MaterialTheme.colorScheme.surface)
            .border(1.dp, accent.copy(alpha = 0.15f), RoundedCornerShape(14.dp))
            .padding(12.dp),
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Text(a.icon, style = MaterialTheme.typography.titleLarge)
        Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(a.title, style = MaterialTheme.typography.titleSmall)
            Text(
                a.body,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}

@Composable
private fun ControlRow(c: AboutControl, accent: Color) {
    Row(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .background(MaterialTheme.colorScheme.surface)
            .border(1.dp, accent.copy(alpha = 0.15f), RoundedCornerShape(14.dp))
            .padding(12.dp),
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Text(c.icon, style = MaterialTheme.typography.titleLarge)
        Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(c.title, style = MaterialTheme.typography.titleSmall)
            Text(
                c.body,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}

@Composable
private fun BuildRow(b: AboutBuild, accent: Color) {
    Column(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .background(MaterialTheme.colorScheme.surface)
            .border(1.dp, accent.copy(alpha = 0.15f), RoundedCornerShape(14.dp))
            .padding(12.dp),
        verticalArrangement = Arrangement.spacedBy(2.dp),
    ) {
        Text(b.name, style = MaterialTheme.typography.titleSmall, color = accent)
        Text(
            b.body,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
        )
    }
}

@Composable
private fun PriceRow(p: AboutPrice, accent: Color) {
    Column(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .background(MaterialTheme.colorScheme.surface)
            .border(1.dp, accent.copy(alpha = 0.15f), RoundedCornerShape(14.dp))
            .padding(12.dp),
        verticalArrangement = Arrangement.spacedBy(3.dp),
    ) {
        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            Text(p.label, style = MaterialTheme.typography.titleSmall)
            Text(p.price, style = MaterialTheme.typography.labelMedium, color = accent)
        }
        Text(
            p.body,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
        )
    }
}
